sdk/schemaregistry/schema-registry-avro/src/schemaRegistryAvroSerializer.ts

```diff
@@ -0,0 +1,226 @@
+// apart this format is inlined here, but handling could be extracted and shared
+// between serializers of different formats in the future.
+//
+// - [4 bytes: Format Indicator]
+//      - Currently always zero to indicate format below.
+//
+// - [32 bytes: Schema ID]
+//      - UTF-8 hexadecimal representation of GUID.
+//      - 32 hex digits, no hyphens.
+//      - Same format and byte order as string from Schema Registry service.
+//      - This will soon be revised to an 8 byte long value with the format
+//        indicator bumped.
+//
+// - [Remaining bytes: Avro payload (in general, format-specific payload)]
+//     - Avro Binary Encoding
+//     - NOT Avro Object Container File, which includes the schema and defeats
+//       the purpose of this serialzer to move the schema out of the message
+//       payload and into the schema registry.
+//
+const FORMAT_INDICATOR = 0;
+const SCHEMA_ID_OFFSET = 4;
+const SCHEMA_ID_LENGTH = 32;
+const PAYLOAD_OFFSET = 36;
+
+interface CacheEntry {
+  /** Schema ID */
+  id: string;
+
+  /** avsc-specific representation for schema */
+  type: avro.Type;
+}
+
+/**
+ * Options for Schema
+ */
+export interface SchemaRegistryAvroSerializerOptions {
+  /**
+   * When true, register new schemas passed to serialize. Otherwise, and by
+   * default, fail if schema has not already been registered.
+   *
+   * Automatic schema registration is NOT recommended for production scenarios.
+   */
+  autoRegisterSchemas?: boolean;
+}
+
+/**
+ * Avro serializer that obtains schemas from a schema registry and does not
+ * pack schemas into its payloads.
+ */
+export class SchemaRegistryAvroSerializer {
+  /**
+   * Creates a new serializer.
+   *
+   * @param registry Schema Registry where schemas are registered and obtained.
+   *                 Usually this is a SchemaRegistryClient instance.
+   *
+   * @param schemaGroup The schema group to use when making requests to the
+   *                    registry.
+   */
+  constructor(
+    registry: SchemaRegistry,
+    schemaGroup: string,
+    options?: SchemaRegistryAvroSerializerOptions
+  ) {
+    this.registry = registry;
+    this.schemaGroup = schemaGroup;
+    this.autoRegisterSchemas = options?.autoRegisterSchemas ?? false;
+  }
+
+  /** The schema group to use when making requests to the registry*/
+  public readonly schemaGroup: string;
+
+  /**
+   * Schema Registry where schemas are registered and obtained.
+   * Usually this is a SchemaRegistryClient instance.
+   */
+  public readonly registry: SchemaRegistry;
+
+  /**
+   * When true, register new schemas passed to serialize. Otherwise, and by
+   * default, fail if schema has not already been registered.
+   *
+   * Automatic schema registration is NOT recommended for production scenarios.
+   */
+  public autoRegisterSchemas: boolean;
+
+  // REVIEW: signature.
+  //
+  // - Better to serialize into a stream? I aborted that for now as I wanted to
```

Comment:
Yeah it's a little weird to depend on `Buffer` if you ever want to support the browser, but I'm not sure what a better solution is for now. Maybe @bterlson knows?